web/src/components/dashboard/HideableTasks.test.tsx

```diff
@@ -0,0 +1,85 @@
+import { HideableTasks } from "./HideableTasks";
+
+jest.mock("@/lib/data-hooks/useUserData", () => ({ useUserData: jest.fn() }));
+jest.mock("@/lib/data-hooks/useRoadmap", () => ({ useRoadmap: jest.fn() }));
+
+const Config = getMergedConfig();
+
+describe("HideableTasks", () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    setupStatefulUserDataContext();
+    useMockRoadmap({});
+  });
+
+  const renderHideableTask = (initialUserData?: UserData) =>
+    render(
+      <WithStatefulUserData initialUserData={initialUserData}>
+        <HideableTasks />
+      </WithStatefulUserData>
+    );
+
+  it("does not render HideableTasks for operating phases that don't display HideableRoadmapTasks", () => {
+    const randomOperatingPhase = randomElementFromArray(
+      OperatingPhases.filter((obj) => obj.displayHideableRoadmapTasks !== true)
+    );
+    useMockUserData({ profileData: generateProfileData({ operatingPhase: randomOperatingPhase.id }) });
+    render(<HideableTasks />);
+
+    expect(screen.queryByText(Config.dashboardDefaults.upAndRunningTaskHeader)).not.toBeInTheDocument();
+  });
+
+  it("renders HideableTasks for operating phases that display HideableRoadmapTasks", () => {
+    const randomOperatingPhase = randomElementFromArray(
+      OperatingPhases.filter((obj) => obj.displayHideableRoadmapTasks === true)
+    );
+    useMockUserData({ profileData: generateProfileData({ operatingPhase: randomOperatingPhase.id }) });
+    render(<HideableTasks />);
+
+    expect(screen.getByText(Config.dashboardDefaults.upAndRunningTaskHeader)).toBeInTheDocument();
+  });
+
+  it("displays number of hidden tasks when hide toggle is clicked", () => {
+    useMockRoadmap({
+      tasks: [generateTask({}), generateTask({})],
+    });
+    const randomOperatingPhase = randomElementFromArray(
+      OperatingPhases.filter((obj) => obj.displayHideableRoadmapTasks === true)
+    );
+
+    const userData = generateUserData({
+      profileData: generateProfileData({ operatingPhase: randomOperatingPhase.id }),
+      preferences: generatePreferences({ isHideableRoadmapVisible: true }),
+    });
+    renderHideableTask(userData);
+
+    fireEvent.click(screen.getByText(Config.dashboardDefaults.hideTaskText));
+
+    expect(screen.getByText("2 Hidden Tasks")).toBeInTheDocument();
+  });
+
+  it("displays roadmap tasks when show toggle is clicked", () => {
+    const randomOperatingPhase = randomElementFromArray(
+      OperatingPhases.filter((obj) => obj.displayHideableRoadmapTasks === true)
+    );
+
+    const userData = generateUserData({
+      profileData: generateProfileData({ operatingPhase: randomOperatingPhase.id }),
+      preferences: generatePreferences({ isHideableRoadmapVisible: false }),
+    });
+    renderHideableTask(userData);
+
+    fireEvent.click(screen.getByText(Config.dashboardDefaults.showTaskText));
+
+    expect(screen.getByTestId("section-plan")).toBeInTheDocument();
```

Comment:
i think this test could also expect that the hidden text is not there. And the test previous could expect that this text is not in the document